src/traces/pie/index.js

```diff
@@ -373,6 +373,12 @@ pie.plot = function(gd, cdpie) {
                     }
                 }
 
+                function handleClick (evt) {
+                    gd._hoverdata = pt;
```

Comment:
Yes please, make it an array. For consistency with other types, as @etpinard says, but also with the converse operation for the same type - displaying hover data - which could still include multiple slices that you want to highlight.